### PR TITLE
Docker Multiarch Build

### PR DESCRIPTION
**Pull request by maxb2**
This adds a multiarch docker build manifest for tagged releases. See my docker repo for an [example](https://hub.docker.com/repository/docker/maxb2/dungeon-revealer/tags?page=1). It makes images for the platforms: `linux/amd64, linux/386, linux/arm/v6, linux/arm/v7, linux/arm64`. This increases the availability of the project to more platforms. `armv6` and `armv7` cover the whole lineup of Raspberry Pi computers, see #256.

There are some serious considerations with this change. The build cannot be run on Travis-CI because it exceeds the time limit of 1 hour. Using Actions with a matrix and some caching, the [build](https://github.com/maxb2/dungeon-revealer/runs/644151932) takes ~1 hour to finish. However, this only runs docker-build on tags or commits to master. it doesn't delay normal CI checks or the binary builds. 

Since this has to be run on Actions, we have to address #364 first. We don't have @apclary's Dockerhub credentials set as secrets on this repo. I would suggest that we acquire the dungeon-revealer username on Dockerhub and then use those credentials. @n1ru4l, can you set secrets in this repo? I can't tell who has what [permissions](https://github.com/orgs/dungeon-revealer/people) here anymore.

**TODO**
 - [x] Use build matrix to reduce duration
 - [x] Set Dockerhub credentials as secrets
 - [x] Update docs
 - [ ] Remove Travis checks from this repo
 - [x] Make a test [tagged release](https://github.com/dungeon-revealer/dungeon-revealer/tree/v1.6.1-multiarch-test) to make sure everything is working properly
   - [x] Remove test tag and discard [release draft](https://github.com/dungeon-revealer/dungeon-revealer/releases/tag/untagged-f5b985845be0c0662244)

### CODE REVIEW

**Comment by n1ru4l** [comment]
It is not possible to use dashes or underscore for the organization name 😖

I claimed https://hub.docker.com/orgs/dungeonrevealer

**Comment by maxb2** [comment]
maxb2 is my docker username as well.

File: .github/workflows/build-docker.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      - name: Available platforms
+        run: echo ${{ steps.buildx.outputs.platforms }}
+
+      - name: Login to DockerHub Registry
+        run: echo ${{ secrets.DOCKER_PASSWORD }} | docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin
+
+      - name: Run Buildx
+        run: |
+          docker buildx build \
+            --platform linux/amd64,linux/386,linux/arm/v6,linux/arm/v7,linux/arm64 \
```
Comment (n1ru4l) [inline .github/workflows/build-docker.yml:41]: Could we reduce the duration by using a matrix, similar to the node.js version matrix?

Comment (maxb2) [reply .github/workflows/build-docker.yml:41]: I'll give it a try.

Comment (maxb2) [reply .github/workflows/build-docker.yml:77]: I managed to reduce the duration to ~1 hour! I had to do some caching between the jobs to achieve it.

File: .github/workflows/build-docker.yml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    if: startsWith(github.ref, 'refs/heads/') # Just the branches
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set environment
+        run: |
+          echo "::set-env name=DOCKER_REPO::${{ secrets.DOCKER_USERNAME }}/dungeon-revealer"
```
Comment (n1ru4l) [inline .github/workflows/build-docker.yml:57]: I think we must distinguish between DOCKER_USERNAME and DOCKER_REPOSITORY_NAME?  As the user name for login is different from the organization name.

File: .github/workflows/build-ci.yml [hunk 3]
```diff
@@ -35,5 +35,5 @@ jobs:
       - name: Build image
         uses: docker/build-push-action@v1
         with:
-          repository: apclary/dungeon-revealer
+          repository: ${{ secrets.DOCKER_USERNAME }}/dungeon-revealer
```
Comment (n1ru4l) [inline .github/workflows/build-ci.yml:38]: We should probably also update this to `dungeonrevealer`, just to be consistent.

Comment (maxb2) [reply .github/workflows/build-ci.yml:38]: Ah, that one slipped through the cracks.

**Comment by n1ru4l** [comment]
Ok I added the secrets

**Comment by maxb2** [comment]
The [test tag](https://github.com/dungeon-revealer/dungeon-revealer/tree/v1.6.1-multiarch-test) is [running](https://github.com/dungeon-revealer/dungeon-revealer/actions/runs/95643819). Once, we see that the Dockerhub repo is good, then we can merge.

**Comment by maxb2** [comment]
This works as intended on a raspi 4 but not on a 3. The raspi 3 is pulling down the `amd64` image. There might be something wrong with the manifest.

**Comment by n1ru4l** [comment]
> the distro repo installed an outdated version of docker. We need to add a minimum version of docker to the docs.

This makes me remember a situation at a former employer where we convinced the sysadmin for using docker for a project. He installed it from the distro repo and it was hella outdated.

I think most people install docker according to the (official) docker documentation and not from some kind of a distro. So just adding a notice should be enough!

@maxb2 Is this now ready for getting merged?

**Comment by maxb2** [comment]
I'm trying one more thing [here](https://github.com/maxb2/dungeon-revealer/runs/646772668), then it will be ready. 

I'm also thinking about caching the sqlite builds somehow. It will be in a different PR though.

**Comment by n1ru4l** [comment]
Do I understand correctly that the 45min duration is mostly compiling sqlite? :D

**Comment by maxb2** [comment]
> Do I understand correctly that the 45min duration is mostly compiling sqlite? :D

At least half of the duration 😖

**Pull request merged**